### PR TITLE
docs: Drop "Platform" from product name.

### PR DESCRIPTION
**Pull request by blp**
Also apply Lalith's suggested changes to concepts.md, and refill paragraphs there.

Suggested by Lalith.

Is this a user-visible change (yes/no): no

### CODE REVIEW

File: docs/what-is-feldera.md [hunk 1]
```diff
@@ -4,18 +4,17 @@ slug: /
...
-Feldera **Continuous Analytics Platform**, also known as
-**Feldera Platform**, is a set of containerized services packaging a fast
-data computational engine for **continuous analytics**. Feldera Platform allows
+Feldera is a set of containerized services packaging a fast
+data computational engine for **continuous analytics**. Feldera allows
 users to run continuous queries directly on data in motion, without storing
 the data in databases or storage systems, before querying.
 
-With Feldera Platform, DML and DDL SQL written for popular databases can operate
+With Feldera, DML and DDL SQL written for popular databases can operate
```
Comment (lalithsuresh) [inline docs/what-is-feldera.md:12]: We don't have DML do we? @ryzhyk

Comment (ryzhyk) [reply docs/what-is-feldera.md:12]: we don't

Comment (mihaibudiu) [reply docs/what-is-feldera.md:12]: The compiler does, and that's how it's tested. And that can be the basis of a cli

Comment (lalithsuresh) [reply docs/what-is-feldera.md:12]: Let's drop the DML mention. If it's not usable from the offered product, it shouldn't be in the documentation.

Comment (blp) [reply docs/what-is-feldera.md]: OK, I changed "DML and DDL SQL" to just "SQL".

**Pull request merged**